db/memtable.cc

```diff
@@ -929,6 +929,16 @@ void MemTable::MultiGet(const ReadOptions& read_options, MultiGetRange* range,
 
     if (found_final_value) {
       iter->value->PinSelf();
+      range->AddValueSize(iter->value->size());
+      if (range->GetValueSize() > read_options.value_size_soft_limit) {
```

Comment:
Should this block be after ```range->MarkKeyDone(iter)``` is called? Otherwise, we'll discard the value for this key.

Reply:
I am not marking the key done whose size exceeds the value_size here so that its status can be set to Abort  in the iteration of marking remaining keys to Aborted. But its value is still added to the range->AddValueSize.

Logic:  So for eg if 4th keys value exceeds the value_size limit then its value is still added  to make the range->GetValueSize() exceeds but its status marked Abort.  This way in DBImpl::MultiGet, I know that value_size_limit has already exceeded as range->GetValueSize() > read_options.value_size_soft_limit. If I don't exceed the range->value_size_ then I can't exit from the loop in the DBMultiGet::MultiGet as I don't know if limit has exceeded or not.

Reply:
In that case, should the option name be change to ```value_size_limit```, since its a hard limit rather than a soft limit.